nomad/structs/keyring.go

```diff
@@ -84,15 +95,17 @@ type RootKeyMeta struct {
 	CreateIndex uint64
 	ModifyIndex uint64
 	State       RootKeyState
+	PublishTime int64
 }
 
 // RootKeyState enum describes the lifecycle of a root key.
 type RootKeyState string
 
 const (
-	RootKeyStateInactive RootKeyState = "inactive"
-	RootKeyStateActive                = "active"
-	RootKeyStateRekeying              = "rekeying"
+	RootKeyStateInactive     RootKeyState = "inactive"
+	RootKeyStateActive                    = "active"
+	RootKeyStateRekeying                  = "rekeying"
+	RootKeyStatePrepublished              = "prepublished"
```

Comment:
the api also has a `deprecated` state, do we not want to mirror it here?

Reply:
We do below. It's hidden by GitHub's interface, but if you extend the diff to lines L100/R113 you'll see it. It's split out because it is, ironically, deprecated so there's a comment to that effect.

Reply:
🤦‍♂️ sorry, missed it.